WooCommerce/Classes/ViewRelated/Themes/ThemesCarouselView.swift

```diff
@@ -0,0 +1,130 @@
+    let themesInfo: [(name: String, url: String)] = [
+        (name: "Google", url: "google.com"),
+        (name: "Facebook", url: "facebook.com"),
+        (name: "WordPress", url: "wordpress.com"),
+        (name: "Microsoft", url: "microsoft.com"),
+        (name: "Amazon", url: "amazon.com"),
+        (name: "Apple", url: "apple.com")
+    ]
+
+    /// The title of the message at the end of the carousel.
+    let lastMessageHeading: String
+
+    /// The content of the message at the end of the carousel.
+    let lastMessageContent: String
+
+
+    /// Scale of the view based on accessibility changes
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    var body: some View {
+        ScrollView(.horizontal, showsIndicators: false) {
+            HStack() {
+
+                // Demo for correct state
+                ForEach(themesInfo, id: \.name) { theme in
+                    if let themeUrl = getThemeUrl(themeUrl: theme.url) {
+                        ThemeImage(url: themeUrl)
+                    }
+                }
+
+                // Demo for error state
+                ThemeName(name: themesInfo[0].name)
+
+                // Demo for last message content
+                VStack {
+                    Text(lastMessageHeading)
+                        .bold()
+                        .bodyStyle()
+                        .padding(.horizontal, Layout.contentPadding)
+
+                    Spacer()
+                        .frame(height: Layout.contentPadding)
+
+                    Text(lastMessageContent)
+                        .subheadlineStyle()
+                            .multilineTextAlignment(.center)
+                        .padding(.horizontal, Layout.contentPadding)
+                }.frame(
+                    width: Layout.imageWidth * scale,
+                    height: Layout.imageHeight * scale
+                )
+            }
+        }
+    }
+}
+
+private struct ThemeImage: View {
+    let url: URL
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    var body: some View {
+        KFImage(url)
+            .resizable()
+            .aspectRatio(contentMode: .fit)
+            .frame(width: Layout.imageWidth * scale, height: Layout.imageHeight * scale)
+            .cornerRadius(Layout.cornerRadius)
+            .shadow(radius: Layout.shadowRadius, x: 0, y: Layout.shadowYOffset)
+            .padding(.init(
+                top: Layout.imageVerticalPadding,
+                leading: Layout.imageHorizontalPadding,
+                bottom: Layout.imageVerticalPadding,
+                trailing: Layout.imageHorizontalPadding)
+            )
+    }
+}
+
+private struct ThemeName: View {
+    let name: String
+
+    var body: some View {
+        VStack {
+            Text(name)
+        }
+        .frame(width: Layout.imageWidth, height: Layout.imageHeight)
+        .background(Color.withColorStudio(
+            name: .wooCommercePurple,
+            shade: .shade0))
```

Comment:
Not sure about this color, just picking a nice color since there is no design for it at the moment.

Reply:
Asked in figma